server/services/v1/api.go

```diff
@@ -137,17 +140,17 @@ func (s *apiService) CommitTransaction(ctx context.Context, r *api.CommitTransac
 		return nil, err
 	}
 
-	tx, err := s.txMgr.GetTx(r.TxCtx)
-	if err != nil {
-		return nil, err
+	session := s.querySessExecutor.GetQuerySession(r.TxCtx.GetId())
+	if session == nil {
+		return nil, api.Errorf(codes.NotFound, "session not found")
 	}
+	defer s.querySessExecutor.RemoveQuerySession(session.txCtx.Id)
```

Comment:
Looks like a lot of duplication. Can it be `s.sessions.Create(...)` ?
`s.sessions.Get(...)`, etc...

Reply:
yeah sure.